ui/dialogs/selectplaylist.go

```diff
@@ -0,0 +1,152 @@
+package dialogs
+
+import (
+	// "fmt"
+	"fmt"
+	"log"
+	"strings"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+	"github.com/deluan/sanitize"
+	"github.com/dweymouth/supersonic/backend/mediaprovider"
+	"github.com/dweymouth/supersonic/sharedutil"
+	"github.com/dweymouth/supersonic/ui/util"
+)
+
+type SelectPlaylist struct {
+	SearchDialog   *SearchDialog
+	mp             mediaprovider.MediaProvider
+	loggedInUser   string
+	allPlaylists   []*mediaprovider.Playlist
+	SkipDuplicates bool
+}
+
+func NewSelectPlaylistDialog(mp mediaprovider.MediaProvider, im util.ImageFetcher, loggedInUser string) *SelectPlaylist {
+
+	sp := &SelectPlaylist{
+		mp:             mp,
+		loggedInUser:   loggedInUser,
+		SkipDuplicates: false,
+	}
+	sd := NewSearchDialog(
+		im,
+		"Select playlist",
+		sp.onSearched,
+		sp.onUpdateSearchResult,
+		sp.onInit,
+	)
+	sp.SearchDialog = sd
+	return sp
+}
+
+func (sp *SelectPlaylist) onInit() ([]*mediaprovider.SearchResult, *widget.Check) {
+	var results []*mediaprovider.SearchResult
+	playlists, err := sp.mp.GetPlaylists()
+	if err != nil {
+		// TODO: surface this error to user
+		log.Printf("error getting playlists: %s", err.Error())
+		return results, nil
+	}
+	sp.allPlaylists = sharedutil.FilterSlice(playlists, func(playlist *mediaprovider.Playlist) bool {
+		return playlist.Owner == sp.loggedInUser
+	})
+	for _, playlist := range sp.allPlaylists {
+		results = append(results, &mediaprovider.SearchResult{
+			Name:       playlist.Name,
+			ID:         playlist.ID,
+			CoverID:    playlist.CoverArtID,
+			Type:       mediaprovider.ContentTypePlaylist,
+			Size:       playlist.TrackCount,
+			ArtistName: playlist.Name,
+		})
+	}
+	skipDuplicatesCheck := widget.NewCheck("Skip duplicates", func(checked bool) {
```

Comment:
you can probably use NewCheckWithData here so that the state of the checkbox is automatically kept in sync with the `sp.SkipDuplicates` bool